lacommunaute/openid_connect/tests/tests_model.py

```diff
@@ -0,0 +1,143 @@
+        self.assertFalse(User.objects.filter(email=proc_user_data.email).exists())
+
+        now = timezone.now()
+        with mock.patch("django.utils.timezone.now", return_value=now):
+            user, created = proc_user_data.create_or_update_user()
+        self.assertTrue(created)
+        self.assertEqual(user.email, OIDC_USERINFO["email"])
+        self.assertEqual(user.last_name, OIDC_USERINFO["usual_name"])
+        self.assertEqual(user.first_name, OIDC_USERINFO["given_name"])
+        self.assertEqual(user.username, OIDC_USERINFO["sub"])
+        self.assertEqual(user.identity_provider, IdentityProvider.PRO_CONNECT)
+
+        self.assertEqual(ForumProfile.objects.get(user=user).user, user)
+
+    def test_update_user_from_user_info(self):
+        USERINFO = {
+            "first_name": "Jeff",
+            "last_name": "BUCKLEY",
+            "email": "jeff@buckley.com",
+            "username": "af6b26f9-85cd-484e-beb9-bea5be13e30f",
+        }
+        user = UserFactory(**(USERINFO))
+        ForumProfile.objects.create(user=user)
+
+        proc_user_data = OIDConnectUserData.from_user_info(OIDC_USERINFO)
+
+        now = timezone.now()
+        with mock.patch("django.utils.timezone.now", return_value=now):
+            _, created = proc_user_data.create_or_update_user()
+
+        self.assertFalse(created)
+
+        user = User.objects.get(username=USERINFO["username"])
+        self.assertEqual(user.first_name, OIDC_USERINFO["given_name"])
+        self.assertEqual(user.last_name, OIDC_USERINFO["usual_name"])
+        self.assertEqual(user.email, OIDC_USERINFO["email"])
+        self.assertNotEqual(user.email, USERINFO["email"])
+        self.assertEqual(user.identity_provider, IdentityProvider.PRO_CONNECT)
+
+        self.assertEqual(1, User.objects.count())
+        self.assertTrue(ForumProfile.objects.get(user=user))
+
+    def test_get_existing_user_with_existing_email(self):
+        """
+        If there already is an existing django user with email OpenID_ sent us, we do not create it again,
+        We use it and we update it with the data form
+        """
+        proc_user_data = OIDConnectUserData.from_user_info(OIDC_USERINFO)
+        UserFactory(
+            first_name=proc_user_data.first_name,
+            last_name=proc_user_data.last_name,
+            email=proc_user_data.email,
+            username=proc_user_data.username,
+        )
+        user, created = proc_user_data.create_or_update_user()
+        self.assertFalse(created)
+        self.assertEqual(user.last_name, OIDC_USERINFO["usual_name"])
+        self.assertEqual(user.first_name, OIDC_USERINFO["given_name"])
+        self.assertEqual(user.username, OIDC_USERINFO["sub"])
+        self.assertEqual(user.identity_provider, IdentityProvider.PRO_CONNECT)
+
+    def test_get_existing_unmigrated_user_with_existing_email(self):
+        """
+        If there already is an existing django user with email sent us,
```

Comment:
S'il est `with another identity provider` il n'est pas `django user` ;)

Reply:
a user existing in django ^^